op-node-derivers.md

```diff
@@ -0,0 +1,554 @@
+- Slow unsafe-block processing stalls the consolidation process for safe-head progression.
+- We also interleave sequencing and derivation steps, whereas these can be parallel.
+- We lock up all execution when reading the chain-state due to consistency needs.
+- There is no ability to attach a new validation process to the driver,
+  making engine-control sharing unnecessarily difficult.
+
+Stylistically there is more to improve too: the `Driver` is a *struct of 29 properties*, all managed in one place.
+
+#### Derivation tech-debt
+
+The `DerivationPipeline` is overloaded with more than just L1 to L2 data transformation:
+- The safe-block changes that happen are cached for later finality signals to work.
+  (More state that derivation has to deal with.) 
+- A plasma "input fetcher" may override fetching/traversal/finality as special case.
+  (Special hooks, rather than substituting what a rollup does.)
+- The EngineQueue derivation stage hosts a payloads queue that is not related to deriving from L1.
+  (No prioritization control as caller.)
+- The EngineQueue wraps an `EngineController`, rather than producing outputs.
+  (Derivation-caller has no control of what/when things are processed. Context timeouts make little sense.)
+- The derivation is started / reset with a "find sync start" algorithm.
+  (Taking a long time to run, not replaceable, and quite fragile.)
+
+And the above bloats the code-path that is fault-proven;
+the unsafe-block and finalization parts are dead code in the fault-proof context.
+
+### Interop
+
+With interop, we are adding an entirely new block safety type:
+["cross-unsafe"](https://github.com/ethereum-optimism/specs/blob/main/specs/interop/verifier.md#cross-unsafe-inputs).
+
+This requires the op-node to:
+- asynchronously validate "unsafe" blocks to become "cross-unsafe".
+- block "safe" progression until blocks are "cross-unsafe" first (applicable to L1-derived block attributes also).
+- maintain a view of external L2 chains, and reorg local L2 chain-state when remote L2 chain-state is invalidated.
+
+### Alt-DA
+
+To support alt-DA and other customizations,
+there is a desire to extend the driver and derivation behavior with a bundled interface.
+
+Adding or modifying functionality should not require many sparse changes.
+Rather, we want a more plugin-like interface, that allows something to fit in and override core functionality,
+while reusing default functionality where desired, and still being structured to be fault-provable.
+
+Extending the stack should not require duplication of the state-transition 
+modifications between the proof system and regular node.
+
+Alt-DA also changes the meaning of "inputs";
+DA commmitments alone are not sufficient to finalize data as we would with rollup data,
+as the underlying data is not finalzied when the L1-registration of the commitment is.
+With plasma, a challenge period has to expire first.
+In other words, alt-DA changes the finality system.
```

Comment:
It really becomes an issues in alt-DA because we wan to use data before it becomes safe.
Spec'd here: https://github.com/ethereum-optimism/design-docs/blob/8d66ce7bd1ec09f9d4f6ebb7765844b03440ed36/plasma-mode.md#finalization

Reply:
Will add the reference to the specs, thanks.